### ISSUE TITLE
Segment visual order gets out of sync with index order

### ISSUE THREAD

**Issue by reganApe** [yonat/MultiSelectSegmentedControl#17]
Something may have changed in how the underlying UISegmentedControl works but I am now having the following problem/

If you start with MultiSelectSegmentedControl outlet in interface builder all the initial UISegment frame.origin.x are all zero, so initSortedSegmentsArray doesn't do much - just results in the same order as the subviews were added.

But if you now add an extra segment programatically it will get a non-zero frame.origin.x and now sortedSegments is no longer in sync with the subview order.

The segments will display as expected but when you try and setSelectedSegmentIndexes: with index 0 visually this will result in the second segment being selected.

**Comment by reganApe** [comment]
Actually its the other way around, the UISegments from IB do have the correct frame, but the ones added programatically are all zero frame.origin

**Comment by reganApe** [comment]
It looks like this is related to auto layout. If new segments are added they have a zero origin until the layout runs, so initSortedSegmentsArray runs to early and gets a corrupt order. If I force auto layout to run then manually recall initSortedSegmentsArray then it works around the issue.

```
        [self.segmentedControl setNeedsLayout];
        [self.segmentedControl layoutIfNeeded];
        [self.segmentedControl initSortedSegmentsArray]; // Made this public...
```

**Comment by yonat** [comment]
I'm not able to reproduce this behavior. Can you show me the code?

Anyway, a better fox would be to add this override to `MultiSelectSegmentedControl.m`:

```
- (void)layoutSubviews
{
    [super layoutSubviews];
    [self initSortedSegmentsArray];
}
```

**Comment by reganApe** [comment]
With that solution the segments will still be incorrectly ordered until after the layout pass.

To reproduce the behaviour just insert a new segment between existing segments with `insertSegmentWithTitle:atIndex:` then immediately call `setSelectedSegmentIndexes:` to select that segment. The previous segment at that index will be selected instead.

**Comment by reganApe** [comment]
Were you able to reproduce with that?

**Comment by yonat** [comment]
Didn't have time for it yet, sorry.

**Comment by yonat** [comment]
Hi @reganApe , I can't reproduce the problem:
If I add segments programatically, then tap on segments -- I still get the correct (new) indexes tapped.

What are you seeing? Can you attach the code, and the steps to reproduce the problem?

**Comment by reganApe** [comment]
You need to programatically set the selected segment too.

So start off with a control containing two segments then later on do this.

```
[segmentedControl insertSegmentWithTitle:@"select me" atIndex:1 animated:NO];
[segmentedControl setSelectedSegmentIndexes:[NSIndexSet indexSetWithIndex:1]];
```

The segment which actually ends up selected will be the one previously occupying position 1, not the newly inserted one.

**Comment by yonat** [comment]
Fixed in 1.1.3 .

**Comment by reganApe** [comment]
Nice one, thanks
